packages/@aws-cdk/cfnspec/build-tools/spec-diff.ts

```diff
@@ -146,15 +149,38 @@ async function main() {
     }
   }
 
-  function pushDownFirstAdditions(update: any) {
-    for (const key of Object.keys(update)) {
-      const added = isAdded(key);
-      if (!added) { continue; }
-      const data = update[key];
-      delete update[key];
-      update[added] = {};
-      for (const subKey of Object.keys(data)) {
-        update[added][`${subKey}__added`] = data[subKey];
+  /**
```

Comment:
Over time, we've built up a lot of logic here in spec-diff.ts but it lacks unit tests. We should consider an item in the future to refactor and tests to it.

Not needed as part of this change.

Reply:
Opened an issue - https://github.com/aws/aws-cdk/issues/6198